payjoin/src/send/mod.rs

```diff
@@ -288,7 +288,10 @@ impl Sender {
         }
 
         match self.extract_rs_pubkey() {
-            Ok(rs) => self.extract_v2(ohttp_relay, rs),
+            Ok(_rs) => {
+                let (req, context_v2) = self.extract_v2(ohttp_relay)?;
+                Ok((req, Context::V2(context_v2)))
+            }
```

Comment:
I wonder if it even makes sense to have this match `extract_highest_version` since it still requires a match on the resulting `Context` to handle both processing v1 and v2

It may be simpler to just have the downstream implementor switch on whether or not extract_v2 errors, then try extract_v1 (if it's a version error), and otherwise return an error.

Reply:
I agree that the try/catch pattern on extract_v2 seems cleaner.